tests/utils/hash_test_suite.cpp

```diff
@@ -0,0 +1,281 @@
+        xlnt::font font2;
+        xlnt_assert_equals(hasher(font1), hasher(font2));
+
+        // Test fonts with different names
+        xlnt::font font3;
+        font3.name("Arial");
+        xlnt_assert(hasher(font1) != hasher(font3));
+
+        // Test fonts with different sizes
+        xlnt::font font4;
+        font4.size(14.0);
+        xlnt_assert(hasher(font1) != hasher(font4));
+
+        // Test bold fonts
+        xlnt::font font5;
+        font5.bold(true);
+        xlnt_assert(hasher(font1) != hasher(font5));
+
+        // Test fonts with colors
+        xlnt::font font6;
+        font6.color(xlnt::color::red());
+        xlnt_assert(hasher(font1) != hasher(font6));
+    }
+
+    void test_hash_uniqueness()
+    {
+        std::hash<xlnt::color> color_hasher;
+        std::hash<xlnt::font> font_hasher;
+        
+        // Test that hash functions work well in practice with unordered_set
+        std::unordered_set<xlnt::color> color_set;
+        std::unordered_set<xlnt::font> font_set;
+        
+        // Test a reasonable set of different colors
+        std::vector<xlnt::color> test_colors = {
+            xlnt::color::red(), xlnt::color::blue(), xlnt::color::green(),
+            xlnt::color::black(), xlnt::color::white(), xlnt::color::yellow(),
+            xlnt::color(xlnt::indexed_color(1)), xlnt::color(xlnt::indexed_color(2)),
+            xlnt::color(xlnt::theme_color(1)), xlnt::color(xlnt::theme_color(2))
+        };
+        
+        // Add some tinted colors
+        xlnt::color tinted_red = xlnt::color::red();
+        tinted_red.tint(0.5);
+        test_colors.push_back(tinted_red);
+        
+        xlnt::color tinted_blue = xlnt::color::blue();
+        tinted_blue.tint(-0.3);
+        test_colors.push_back(tinted_blue);
+        
+        // Insert all colors and verify they are properly handled
+        for (const auto& color : test_colors) {
+            color_set.insert(color);
+        }
+        
+        // We should have the same number of colors in the set as we inserted
+        // (since they're all different)
+        xlnt_assert_equals(color_set.size(), test_colors.size());
+        
+        // Test different font combinations
+        std::vector<std::string> font_names = {"Arial", "Calibri", "Times New Roman"};
+        std::vector<double> font_sizes = {10.0, 12.0, 14.0};
+        std::vector<bool> bold_options = {false, true};
+        
+        std::size_t expected_font_count = 0;
+        for (const auto& name : font_names) {
+            for (const auto& size : font_sizes) {
+                for (bool bold : bold_options) {
+                    xlnt::font font;
+                    font.name(name).size(size).bold(bold);
+                    font_set.insert(font);
+                    expected_font_count++;
+                }
+            }
+        }
+        
+        // We should have all the fonts we created
+        xlnt_assert_equals(font_set.size(), expected_font_count);
```

Comment:
This does not really test your hash function. An unordered set may contain multiple values with the same hash as long as they are different according to the equality operator.